internal/provider/job_resource.go

```diff
@@ -1219,9 +1219,10 @@ func (r *JobResource) Schema(ctx context.Context, req resource.SchemaRequest, re
 				NestedObject: schema.NestedAttributeObject{
 					Attributes: map[string]schema.Attribute{
 						"id": schema.StringAttribute{
-							Description: "The unique identifier of the destination resource. This is set after creation",
-							Optional:    true,
-							Computed:    true,
+							Description:   "The unique identifier of the destination resource. This is set after creation",
+							Optional:      true,
+							Computed:      true,
+							PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
```

Comment:
This fixes the issue where the id was not present in the planned state which was not allowing updates to the destinations. Unclear why this was not being caught before.